rasa/core/policies/policy.py

```diff
@@ -176,10 +179,17 @@ def _create_featurizer(cls, policy_config: Dict[Text, Any]) -> TrackerFeaturizer
                 **state_featurizer_config
             )
 
-        return featurizer_func(**featurizer_config)
-
-    @staticmethod
-    def _standard_featurizer() -> MaxHistoryTrackerFeaturizer:
+        featurizer = featurizer_func(**featurizer_config)
+        if (
+            isinstance(featurizer, MaxHistoryTrackerFeaturizer)
+            and POLICY_MAX_HISTORY in policy_config
+            and POLICY_MAX_HISTORY not in featurizer_config
+        ):
+            featurizer.max_history = policy_config[POLICY_MAX_HISTORY]
+        return featurizer
+
+    def _standard_featurizer(self) -> MaxHistoryTrackerFeaturizer:
+        """Initializes the standard featurizer for this policy."""
```

Comment:
I think that logic doesn't quite work as Memoization doesn't use SingleStateFeaturizer. However, it was possible to make it a bit more simple by making the Policy class have the behavior that uses the config max history too so that way it was identical to TED and TED's method could be removed.

Reply:
you could obviously make it even more abstract by definining a standard tracker feautrizer class and a standard state featurizer class per component, then you could make their behavior all equal - not sure it's worth it at this point though